Source/WebCore/Modules/webaudio/AudioWorkletProcessorConstructor.idl

```diff
@@ -28,5 +28,4 @@
 
 [
     Conditional=WEB_AUDIO,
-    GenerateIsReachable=ImplScriptExecutionContext
```

Comment:
It seems this has no effect on callbacks, shouldn't we fix that instead? If I look at the generated code for JSAudioWorkletProcessorConstructor, it definitely doesn't generate the `isReachableFromOpaqueRoots` function I'd expect it to. I suspect the bindings generator ignores `GenerateIsReachable` for callbacks and only considers it for interfaces.

One of the reasons I'm concerned is:
- how about IdleRequestCallback, doesn't it have the same bug for example? It also uses `GenerateIsReachable=ImplScriptExecutionContext` and the generated bindings code also doesn't contain a isReachableFromOpaqueRoots function.

Reply:
The code for `GenerateIsReachable` for callbacks is a bit different from interfaces. For callbacks all it does is it constructs the `JSC::WeakImpl` with a pointer to the `ScriptExecutionContext` instead of a pointer to `this` (the callback). In the main thread the script execution context has been added as an opaque root to the visitor so it works there. Since its not working here and I went back to make sure the code generator was working like I meant it to it must be that nothing is adding the AudioWorklet's context as an opaque root to the visitor. So that is what probably needs to be fixed here.

Reply:
For completeness, the `isReachableFromOpaqueRoots` implementation is in `WebCore/bindings/js/JSCallbackData.h` and so the code generator doesn't need to generate the WeakHandleOwner stuff  like it does for other IDL types.

Reply:
Yeah so adding `addWebCoreOpaqueRoot(visitor, wrapped())` to `JSAudioWorkletGlobalScope::visitAdditionalChildren` fixes this with no need to lock the map. Since its not possible to unregister a processor constructor after registering (as far as I can tell) that one line change is all that's needed to fix the bug.